R/zchunk_L225.hydrogen.R

```diff
@@ -1,237 +1,287 @@
+        L225.SubsectorShrwtFllt_h2
+    }
+
+    if(exists("A25.subsector_interp")){
+      # L225.SubsectorInterp_h2 and L225.SubsectorInterpTo_h2: Subsector shareweight interpolation of hydrogen sectors
+      if(any(is.na(A25.subsector_interp$to.value))){
+        A25.subsector_interp[is.na(A25.subsector_interp$to.value),] %>%
+          write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorInterp']], GCAM_region_names)->
+          L225.SubsectorInterp_h2
+      }
+      if(any(!is.na(A25.subsector_interp$to.value))){
+        A25.subsector_interp[!is.na(A25.subsector_interp$to.value),] %>%
+          write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorInterpTo']], GCAM_region_names) ->
+          L225.SubsectorInterpTo_h2
+      }
+    }
+
+    # 1c. Technology information
+
+    # L225.StubTech_h2: Identification of stub technologies of hydrogen
+    # Note: assuming that technology list in the shareweight table includes the full set (any others would default to a 0 shareweight)
+    A25.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[['Tech']], GCAM_region_names) -> L225.StubTech_h2
+    names(L225.StubTech_h2) <- LEVEL2_DATA_NAMES[['StubTech']]
+
+    # L225.GlobalTechEff_h2: Energy inputs and efficiencies of global technologies for hydrogen
+    # Efficiencies of global technologies
+    A25.globaltech_eff %>%
+      gather(year, efficiency, -supplysector, -subsector, -technology, -minicam.energy.input) %>%
+      mutate(year=as.numeric(year)) ->
+      A25.globaltech_eff_raw_long
+
+    df_years <- unique(A25.globaltech_eff_raw_long$year)
+    int_years <- c(BASE_YEARS, FUTURE_YEARS)[which(!(c(BASE_YEARS, FUTURE_YEARS) %in% df_years))]
+
+    A25.globaltech_eff_raw_long %>%
+      select(-efficiency,-year) %>%
+      unique() %>%
+      repeat_add_columns(tibble::tibble(year=int_years)) %>%
+      bind_rows(A25.globaltech_eff_raw_long) %>%
+      group_by(supplysector, subsector, technology, minicam.energy.input) %>%
+      mutate(efficiency=approx_fun(year, efficiency, rule=1)) %>%
+      mutate(efficiency=round(efficiency,energy.DIGITS_EFFICIENCY)) %>%
+      filter(year %in% c(BASE_YEARS,FUTURE_YEARS)) %>%
+      rename(sector.name=supplysector) %>% #Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(subsector.name=subsector) ->
+      L225.GlobalTechEff_h2
+
+    # L225.GlobalTechCost_h2: Costs of global technologies for hydrogen
+    # Costs of global technologies
+    A25.globaltech_cost %>%
+      gather(year, input.cost, -supplysector, -subsector, -technology, -minicam.non.energy.input) %>%
+      mutate(year=as.numeric(year)) ->
+      A25.globaltech_cost_raw_long
+
+    df_years <- unique(A25.globaltech_cost_raw_long$year)
+    int_years <- c(BASE_YEARS, FUTURE_YEARS)[which(!(c(BASE_YEARS, FUTURE_YEARS) %in% df_years))]
+
+    A25.globaltech_cost_raw_long %>%
```

Comment:
Sure. I'll add more comments. Basically the section is trying to behave like what 'interpolate_and_melt' does in the old data system.